cppsrc/main.cpp

```diff
@@ -109,140 +103,91 @@ Napi::FunctionReference Snowflake::constructor;
  * Takes the current timestamp, last timestamp, sequence, and macID
  * and generates a 64 bit long integer by performing bitwise operations
  * 
- * First 41 bits are filled with current timestamp
- * Next 10 bits are filled with the node/machine id (max size can be 4096)
- * Next 12 bits are filled with sequence which ensures that even if timestamp didn't change the value will be generated
+ * First 11 bits are filled with object id
+ * Next 6 bits are reserved
+ * Next 39 bits are filled with the current timestamp
+ * Next 8 bits are filled with sequence which ensures that even if timestamp didn't change the value will be generated
  * 
  * Function can theorotically generate 1024 unique values within a millisecond without repeating values
 */
-Napi::Value Snowflake::getUniqueIDBigInt(const Napi::CallbackInfo &info)
-{
+Napi::Value Snowflake::getUniqueIDBigInt(const Napi::CallbackInfo &info) {
     auto env = info.Env();
 
     uint64_t currentTimestamp = getCurrentTime() - this->_CUSTOM_EPOCH;
+    this->_objectId = info[0].As<Napi::Number>().Int32Value() & maxObjectId;
 
-    if (currentTimestamp == this->_lastTimestamp)
-    {
+    if (currentTimestamp == this->_lastTimestamp) {
         this->_sequence = (this->_sequence + 1) & maxSequence;
-        if (!this->_sequence)
-        {
-            std::this_thread::sleep_for(std::chrono::milliseconds(1));
+        if (!this->_sequence) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(10));
             ++currentTimestamp;
         }
-    }
-    else
-    {
+    } else {
         this->_sequence = 0;
     }
 
     this->_lastTimestamp = currentTimestamp;
 
     uint64_t id{};
-    id = currentTimestamp << (TOTAL_BITS - EPOCH_BITS);
-    id |= (this->_nodeID << (TOTAL_BITS - EPOCH_BITS - NODE_ID_BITS));
+    id = this->_objectId << (TOTAL_BITS - OBJECT_ID_BITS);
+    id |= (this->_reservedId << (TOTAL_BITS - OBJECT_ID_BITS - RESERVED_BITS)); 
+    id |= (currentTimestamp << (TOTAL_BITS - OBJECT_ID_BITS - RESERVED_BITS - EPOCH_BITS)); 
     id |= this->_sequence;
 
     return Napi::BigInt::New(env, id);
 }
 
 /**
- * Returns timestamp at which the id was generated by retreiving
- * the first 41 bits of the id, which are filled with current timestamp
+ * Returns objectId at which the id was generated by retreiving
+ * the first 11 bits of the id, which are filled with objectId
  * bits
```

Comment:
If we were to rewrite the existing message, it'd go like this:

```suggestion
 * Returns the objectId of the id by retrieving its first 11 bits.
```

However I generally prefer hiding implementation details from the documentation because the implementation might change (e.g. the 11 bit part) and the documentation will be wrong if we forget to update it - so I'd probably just leave it as simple as:

```suggestion
 * Returns the objectId of the id.
```